### DISCUSSION TITLE
recording of filament

### DISCUSSION THREAD

**Discussion by wriel1989** [greghesp/ha-bambulab#534] [Q&A]
Is there already a good way to record filament over time?
Know we have sensor.bambu_x1c_print_length and sensor.bambu_x1c_print_weight.
This is only for current print.

Like to record how much of each type filament and/or color is being used over longer period of time.

Together whit this. create automation for checking if sensor.bambu_x1c_print_length value is not overexceeding of the filament that is left on the spool. 
To directly know if you will run out of filament before the end of the print. 

Make it more easier to cancel it if you do not have more of this type of filament and need to order more.

**Comment by eecharlie** [comment]
Bumping this as I'd love to do filament odometry - keeping track of approx % remaining on each spool. 

It's not sufficient to know print length & weight of current print, you also need to know to the filament source (spool). 
There's a tool called Spoolman out there that has seen some [integration](https://github.com/Disane87/spoolman-homeassistant?tab=readme-ov-file) with HomeAssistant, and a [reddit thread](https://www.reddit.com/r/BambuLab/comments/1hamev5/keeping_spoolman_uptodate_with_bambu_lab_and_home/) in which someone created a HomeAssistant script but it's based on spool revolutions I think.

**Reply by platini76** [reply]
> reddit thread

for me this won'r work... there is none who can integrate bambu update to spoolman?
